File: module_home/src/main/java/com/punuo/pet/home/care/model/CareData.java
package com.punuo.pet.home.care.model;

import com.google.gson.annotations.SerializedName;

/**
 * Created by han.chen.
 * Date on 2019-08-15.
 **/
public class CareData {

    /**
     * icon :
     * label : 洗澡清洁
     * petName :
     * date :
     * needAlarm : false
     * repeatCycle : day
     */

//    @SerializedName("icon")
//    public String icon;
    @SerializedName("carename")
    public String careName;
    @SerializedName("petname")
    public String petName;
    @SerializedName("date")
    public long date;
//    @SerializedName("needAlarm")
//    public boolean needAlarm;
//    @SerializedName("repeatCycle")
//    public String repeatCycle;

    public CareData(String icon,String label,String petName,long date){
//        this.icon = icon;
        this.careName = label;
        this.petName = petName;
        this.date = date;
    }

    public String getIcon(){
        return "";
    }
    public String getCareName(){
        return careName;
    }
    public String getPetName(){
        return petName;
    }
    public long getDate(){
        return date;
    }
}

File: module_home/src/main/java/com/punuo/pet/home/care/model/CareData.kt
package com.punuo.pet.home.care.model

import com.google.gson.annotations.SerializedName
import com.punuo.sys.sdk.model.PNBaseModel

/**
 * Created by han.chen.
 * Date on 2020/12/8.
 **/
class CareData : PNBaseModel() {

    @SerializedName("picture")
    var icon: String? = null

    @SerializedName("carename")
    var careName: String? = null

    @SerializedName("petname")
    var petName: String? = null

    @SerializedName("date")
    var date: Long = 0

    val theme: String?
        get() {
            return when (careName) {
                "洗澡清洁" ,"Cleaning" -> "ShowerPlan"
                "体检" ,"Health Examination"-> "HealthExam"
                "买宠物粮", "Pet Food" -> "PetFood"
                "体内驱虫", "Insect Repellent" -> "InnerInsect"
                "体外驱虫", "Expelling parasite" -> "OuterInsect"
                "疫苗注射", "Vaccination" -> "TakeVaccine"
                "美容护理", "Aesthetic Nursing" -> "MakeBeauty"
                "遛宠", "Walking" -> "WalkPet"
                else -> ""
            }
        }
}